articles/142_idl.md

```diff
@@ -35,3 +35,50 @@ Further it describes how the interfaces are used to generate code in C, C++ and
 <div class="alert alert-warning" markdown="1">
   <b>TODO:</b> All of this too
 </div>
+
+### Type Mapping
+
+IDL types mapped to different types in different languages.
+
+
+| IDL type    | C type      | C++ type    | Python type          |
+| --------    | ----------- | ----------- | -------------------- |
+| bool        | \_Bool      | bool        | bool                 |
+| octet       | uint8\_t    | uint8\_t    | bytes with length 1  |
+| char        | char16\_t   | char16\_t   | str with length 1    |
+| wchar       | signed char | char        | str with length 1    |
+| float       | float       | float       | float                |
+| double      | double      | double      | float                |
+| long double | long double | long double | float                |
+| int8        | int8\_t     | int8\_t     | int                  |
+| uint8       | uint8\_t    | uint8\_t    | int                  |
+| int16       | int16       | int16       | int                  |
+| uint16      | uint16      | uint16      | int                  |
+| int32       | int32       | int32       | int                  |
+| uint32      | uint32      | uint32      | int                  |
+| int64       | int64       | int64       | int                  |
+| uint64      | uint64\_t   | uint64\_t   | int                  |
+
+Unless are the default mappings for array, sequence, and string types unless otherwise specified.
+
+| IDL type                | C type                 | C++ type            | Python type |
+| ----------------------- | ---------------------- | ------------------- | ----------- |
+| array                   | T [FIXED\_SIZE]        | std::array<T, N>    | list        |
+| unbounded sequence      | struct {size\_t, T * } | std::vector<T>      | list        |
+| bounded sequence        | struct {size\_t, T * } | custom\_class<T, N> | list        |
+| string                  | char *                 | std::string         | str         |
+| bounded string          | char *                 | std::string         | str         |
+| wstring                 | char16\_t *            | std::u16string      | str         |
+| bounded wstring         | char16\_t *            | std::u16string      | str         |
+
+These array and sequence types have special mappings.
+If a cell is blank then the default mapping is used.
+
+| IDL type                     | C type                 | C++ type       | Python type |
+| ---------------------------- | ---------------------- | -------------- | ----------- |
+| array of string              | rcutils\_string\_array |                |             |
+| unbounded sequence of string | rcutils\_string\_array |                |             |
+| bounded sequence of string   | rcutils\_string\_array |                |             |
+| array of octet               |                        |                | bytes       |
+| array of char                | char *                 | std::string    | str         |
```

Comment:
It may not make sense to have `array of char` map to `char *` in `C` because that means it couldn't contain a value of zero.

Reply:
array of char and wchar no longer strings in cd1e3b27b1352f83effd43f8b123c23728b04542